operator/bucketcontroller/delete.go

```diff
@@ -55,12 +56,28 @@ func (p *ProvisioningPipeline) deleteAllObjects(ctx *pipelineContext) error {
 		}
 	}()
 
-	for obj := range p.minioClient.RemoveObjects(ctx, bucketName, objectsCh, minio.RemoveObjectsOptions{GovernanceBypass: true}) {
+	bypassGovernance, err := p.isBucketLockEnabled(ctx, bucketName)
+	if err != nil {
+		return fmt.Errorf("cannot determine bucketLockConfig of bucket %s: %w", bucketName, err)
+	}
+
+	for obj := range p.minioClient.RemoveObjects(ctx, bucketName, objectsCh, minio.RemoveObjectsOptions{GovernanceBypass: bypassGovernance}) {
```

Comment:
Not sure if I understand the bug this fixes. So if we specify to bypass governance, but bucket lock is not enabled, we get an error from exoscale, or why can't we unconditionally bypass it?

Reply:
Exactly, if it's not enabled and we set it to `true` it will throw an error and vice-versa.

Reply:
O wow, I'd expect them to ignore the bypass field and delete the objects if we specify it for a bucket without locks..